### PR TITLE
Set edition in template

### LINKED ISSUE #1176: Set edition in template & snforge_std and update imports
Set an edition in forge template to an explicit value (preferably the newest edition), and update code and imports so it works.

### PR DESCRIPTION
**Pull request by drknzz**
Closes #1176 

## Introduced changes

- Set `edition = "2023_10"` in template and snforge_std
- Update snforge_std imports

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/Scarb.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "snforge_std"
 version = "0.1.0"
+edition = "2023_10"
```
Comment (drknzz) [inline snforge_std/Scarb.toml:4]: Should it be added to changelog?

File: crates/forge/src/init.rs [hunk 2]
```diff
@@ -61,6 +63,17 @@ fn add_target_to_toml(path: &Path) -> Result<()> {
     Ok(())
 }
 
+fn set_cairo_edition(path: &Path, cairo_edition: &str) -> Result<()> {
+    let config_file = fs::read_to_string(path)?;
+    let mut doc = config_file.parse::<Document>().expect("invalid document");
```
Comment (piotmag769) [inline crates/forge/src/init.rs:68]: I think this parsing should be moved to `run` or a separate function so we don't parse Scarb.toml twice (here and in `add_target_to_toml`)

File: snforge_std/Scarb.toml (hunk 1 above)
Comment (piotmag769) [reply snforge_std/Scarb.toml:4]: We can I guess

**Review by piotmag769 — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Comment by piotmag769** [comment]
Actually shouldn't we set it explicitly in `sncast_std` too? cc @THenry14

**Comment by THenry14** [comment]
> Actually shouldn't we set it explicitly in `sncast_std` too? cc @THenry14

It makes sense to do this, yes.

**Pull request merged**